### PR TITLE
using a more scoped env fixture, which enables changing nodes creation arguments

### PR DESCRIPTION
**Pull request by osherdp**

### CODE REVIEW

File: discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py [hunk 1]
```diff
@@ -27,6 +27,8 @@ def _create_tf_folder(self):
         tf_folder = utils.get_tf_folder(self.cluster_name)
         logging.info("Creating %s as terraform folder", tf_folder)
         utils.recreate_folder(tf_folder)
+        utils.recreate_folder(os.path.join(tf_folder, "network"))
```
Comment (tsorya) [inline discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py:30]: why? no need for this

Comment (osherdp) [reply discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py:30]: Failed on my env without it

Comment (tsorya) [reply discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py:30]: very strange, need to check

File: discovery-infra/test_infra/helper_classes/cluster.py [hunk 2]
```diff
@@ -199,14 +204,25 @@ def wait_for_install(
...
-        ):
+            self,
+            controller,
+            iso_download_path=None,
+            ssh_key=None,
+            nodes_count=None,
+            vip_dhcp_allocation=None,
+            cluster_machine_cidr=None):
+
+        if iso_download_path is None:
```
Comment (tsorya) [inline discovery-infra/test_infra/helper_classes/cluster.py:215]: why this way is better?

File: discovery-infra/test_infra/helper_classes/cluster.py [hunk 3]
```diff
@@ -199,14 +204,25 @@ def wait_for_install(
...
+            iso_download_path = self._env["iso_download_path"]
```
Comment (tsorya) [inline discovery-infra/test_infra/helper_classes/cluster.py:216]: if it is lets make it :
iso_download_path = iso_download_path or self._env["iso_download_path"]

Comment (tsorya) [reply discovery-infra/test_infra/helper_classes/cluster.py:216]: will look prettier

File: discovery-infra/tests/base_test.py [hunk 4]
```diff
@@ -12,31 +12,39 @@
...
 class BaseTest:
-    @pytest.fixture(scope="function")
-    def node_controller(self, setup_node_controller):
-        controller = setup_node_controller
-        controller.set_correct_boot_order_to_all_nodes()
-        yield controller
-        controller.shutdown_all_nodes()
-        controller.format_all_node_disks()
+    @pytest.fixture()
+    def node_controller(self, node_controller_class):
```
Comment (tsorya) [inline discovery-infra/tests/base_test.py:22]: we deleted nodecontroller fixture, not sure i agree with this one.
It looks much more complicated and for no reason. I like session scope fixture more

File: discovery-infra/tests/base_test.py [hunk 5]
```diff
@@ -62,34 +70,30 @@ def delete_cluster_if_exists(self, api_client, cluster_name):
         if cluster_to_delete:
             api_client.delete_cluster(cluster_id=cluster_to_delete['id'])
 
-    @staticmethod
-    def get_cluster_by_name(api_client, cluster_name):
+    def get_cluster_by_name(self, api_client, cluster_name):
```
Comment (tsorya) [inline discovery-infra/tests/base_test.py:73]: need to delete this one from here and move logic to cluster object

File: discovery-infra/tests/base_test.py [hunk 6]
```diff
@@ -62,34 +70,30 @@ def delete_cluster_if_exists(self, api_client, cluster_name):
...
         return None
 
-    @staticmethod
-    def generate_and_download_image(
-        cluster_id,
-        api_client,
-        iso_download_path=env_variables['iso_download_path'],
-        ssh_key=env_variables['ssh_public_key']
-    ):
+    def generate_and_download_image(self, cluster_id, api_client, iso_download_path=None, ssh_key=None):
```
Comment (tsorya) [inline discovery-infra/tests/base_test.py:80]: same

File: discovery-infra/tests/base_test.py [hunk 7]
```diff
@@ -62,34 +70,30 @@ def delete_cluster_if_exists(self, api_client, cluster_name):
...
+
         logging.info(iso_download_path)
         api_client.generate_and_download_image(
             cluster_id=cluster_id,
             ssh_key=ssh_key,
             image_path=iso_download_path
         )
 
-    @staticmethod
-    def wait_until_hosts_are_registered(
```
Comment (tsorya) [inline discovery-infra/tests/base_test.py:88]: same

File: discovery-infra/tests/base_test.py [hunk 8]
```diff
@@ -100,35 +104,39 @@ def wait_until_hosts_are_registered(
             ]
         )
 
-    @staticmethod
-    def wait_until_hosts_are_discovered(
-        cluster_id,
-        api_client,
-        nodes_count=env_variables['num_masters']+env_variables['num_workers']
-    ):
+    def wait_until_hosts_are_discovered(self, cluster_id, api_client, nodes_count=None):
```
Comment (tsorya) [inline discovery-infra/tests/base_test.py:107]: same

File: discovery-infra/tests/conftest.py [hunk 9]
```diff
@@ -21,54 +21,57 @@
...
 
 @pytest.fixture(scope="session")
-def setup_node_controller():
-    logging.info("Setup node controller")
-    controller = nodeController(**env_variables)
-    controller.prepare_nodes()
-    yield controller
-    logging.info("Teardown node controller")
-    controller.destroy_all_nodes()
+def node_controller_class():
```
Comment (tsorya) [inline discovery-infra/tests/conftest.py:76]: i don't understand why u don't like it? I want session fixture in this case

**Pull request closed without merging**